code/game/objects/items/stacks/nanopaste.dm

```diff
@@ -58,3 +58,63 @@
 					return
 			else
 				user << "<span class='notice'>Nothing to fix in here.</span>"
+
+
+/obj/item/stack/nanopaste/surge
+	name = "modified nanopaste"
+	singular_name = "nanite swarm"
+	desc = "A tube of paste containing swarms of repair nanites. This one appears to contain different nanites."
+	icon = 'icons/obj/nanopaste.dmi'
+	icon_state = "tube-surge"
+	origin_tech = list(TECH_MATERIAL = 4, TECH_ENGINEERING = 4, TECH_MAGNET = 5, TECH_POWER = 5, TECH_COMBAT = 3, TECH_ILLEGAL = 4)
+	amount = 20
+	var/used = FALSE
+	construction_cost = null
+
+/obj/item/stack/nanopaste/surge/attack(mob/living/M as mob, mob/user as mob, var/target_zone)
+	if (!istype(M) || !istype(user))
+		return 0
+
+	if(used)
+		to_chat(user, "<span class='warning'>[src] has depleted it's nanites. </span>")
+		return 0
+
+	if (isipc(M))
+		var/mob/living/carbon/human/H = M
+		if(!H.surge)
+			visible_message(
+			"<span class='notice'> [user] is trying to apply [src] to itself</span>",
+			"<span class='notice'> You start applying [src] to yourself</span>"
+			)
+			if (!do_after(usr, 1 SECONDS, act_target = user))
+				return 0
+			H.surge_left = rand(1, 3)
+			H.surge = TRUE
+			visible_message(
```

Comment:
The visible message should be called as `user.visible_message`.

The arguments for this one are flipped.